### PR TITLE
Diffusion API docs

### PR DESCRIPTION
**Pull request by CharlelieLrt**
# PhysicsNeMo Pull Request

## Checklist

- [ ] I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/physicsnemo/blob/main/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.
- [ ] The [CHANGELOG.md](https://github.com/NVIDIA/physicsnemo/blob/main/CHANGELOG.md) is up to date with these changes.
- [ ] An [issue](https://github.com/NVIDIA/physicsnemo/issues) is linked to this pull request.
- [ ] If I am implementing a new model or modifying any existing model, I have followed the [Models Implementation Coding Standards](https://github.com/NVIDIA/physicsnemo/wiki/Coding-standards-for-models-implementation).

## Review Process

**All PRs are reviewed by the PhysicsNeMo team before merging.**

Depending on which files are changed, GitHub may automatically assign a maintainer for review.

We are also testing AI-based code review tools (e.g., Greptile), which may add automated comments with a confidence score.
This score reflects the AI’s assessment of merge readiness and is **not** a qualitative judgment of your work, nor is
it an indication that the PR will be accepted / rejected.

AI-generated feedback should be reviewed critically for usefulness.
You are not required to respond to every AI comment, but they are intended to help both authors and reviewers.
Please react to Greptile comments with 👍 or 👎 to provide feedback on their accuracy.

### CODE REVIEW

**Comment by ktangsali** [comment]
Depends on https://github.com/NVIDIA/physicsnemo/pull/1471

File: docs/api/models/diffusion_unets.rst [hunk 1]
```diff
@@ -0,0 +1,325 @@
...
+    lead_time_steps = 3  # Maximum supported lead-time is 2 * dt
+    C_LT = 6  # 6 channels for each lead-time embeddings
+
+    # Create a SongUNet with a lead-time embedding grid of shape
+    # (lead_time_steps, C_lt_emb, res, res)
+    model = SongUNetPosLtEmbd(
+        img_resolution=res,
+        in_channels=C_x + C_cond + C_PE + C_LT,  # in_channels must include the number of channels in lead-time grid
+        out_channels=C_x,
+        label_dim=16,
```
[the remark this answers is not in the document]
Comment (CharlelieLrt) [reply docs/api/models/diffusion_unets.rst:224]: Let me decide what we put in the docs, ok?

File: docs/api/diffusion/noise_schedulers.rst [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+Noise schedulers are the central abstraction in the PhysicsNeMo diffusion
+framework.  A noise scheduler defines the forward diffusion process (how noise
+is added to data) and provides the ingredients needed for both training and
+inference.  It is the *only* component that participates in every stage of
+the pipeline: training, loss computation, denoiser construction, and sampling.
+
+If you are familiar with the
+`Scheduler API in HuggingFace Diffusers <https://huggingface.co/docs/diffusers/api/schedulers/overview>`_,
+PhysicsNeMo noise schedulers serve a broadly similar role.  A key design
+difference is that the PhysicsNeMo
```
Comment (megnvidia) [inline docs/api/diffusion/noise_schedulers.rst:17]: wow this noise scheduler does some heavy lifting.

File: docs/api/diffusion/noise_schedulers.rst [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+the coupling between noise schedule and reverse process in one place and avoids
+the need for schedule-specific sampler code.
+
+
+Role in Training
+----------------
+
+During training, the noise scheduler is responsible for three operations:
+
+1. **Sampling diffusion times** via :meth:`~NoiseScheduler.sample_time`.
```
Comment (megnvidia) [inline docs/api/diffusion/noise_schedulers.rst:31]: is the **order** that these happen something you are trying to show?  I am suspecting that making this a bullet list would be better. Also could you replace 'via' with 'using'.  It is part of the Nvidia style guide -guidance

File: docs/api/diffusion/noise_schedulers.rst [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+    loss = loss_fn(x0)   # Internally samples t, adds noise, computes weighted MSE
+    loss.backward()
+
+
+Role in Inference (Sampling)
+----------------------------
+
+During sampling, the noise scheduler provides:
+
+1. **Time-step schedule** via :meth:`~NoiseScheduler.timesteps`.
```
Comment (megnvidia) [inline docs/api/diffusion/noise_schedulers.rst]: same comments here.  Unless these happen in this specific order, just use bullets. Also swap out via

File: docs/api/diffusion/noise_schedulers.rst [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+
+
+Three Levels of Customization
+------------------------------
+
+Following the framework's
+:ref:`design philosophy <diffusion_introduction>`, noise schedulers are
+available at three levels:
+
+**Protocol** (:class:`NoiseScheduler`) --- The minimal interface.  Any object
```
Comment (megnvidia) [inline docs/api/diffusion/noise_schedulers.rst]: you could put these items in a bullet list.  Then instead of the --- you can put :

File: docs/api/diffusion/noise_schedulers.rst [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+that implements the six required methods can be used as a noise scheduler.
+This is the right choice for fully custom forward processes (e.g.,
```
Comment (megnvidia) [inline docs/api/diffusion/noise_schedulers.rst]:
```suggestion
This is the right choice for fully custom forward processes (for example,
```

File: docs/api/diffusion/noise_schedulers.rst [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+non-Gaussian, non-linear, discrete).
+
+**Abstract base class** (:class:`LinearGaussianNoiseScheduler`) --- For the
+common family of linear-Gaussian forward processes of the form
+:math:`\mathbf{x}(t) = \alpha(t)\,\mathbf{x}_0 + \sigma(t)\,\boldsymbol{\epsilon}`.
+This base class implements noise injection, score conversion, and denoiser
+construction.  Subclasses only need to define the schedule-specific quantities:
+:math:`\alpha(t)`, :math:`\sigma(t)`, their derivatives, the inverse mapping,
```
Comment (megnvidia) [inline docs/api/diffusion/noise_schedulers.rst]:
```suggestion

* :math:`\alpha(t)`
* :math:`\sigma(t)`
* their derivatives
* the inverse mapping
```

File: docs/api/diffusion/noise_schedulers.rst [hunk 8]
```diff
@@ -0,0 +1,166 @@
...
+and the discretization.
```
Comment (megnvidia) [inline docs/api/diffusion/noise_schedulers.rst]:
```suggestion
* the discretization
```

File: docs/api/diffusion/preconditioners.rst [hunk 9]
```diff
@@ -0,0 +1,165 @@
...
+
+    All built-in preconditioners are designed so that the preconditioned output
+    is an :math:`\mathbf{x}_0`-prediction (clean data estimate).  They are
+    intended for use with
+    :class:`~physicsnemo.diffusion.metrics.losses.MSEDSMLoss` with
+    ``prediction_type="x0"`` (the default).
+
+See the :class:`BaseAffinePreconditioner` docstring for additional examples,
+including how to write thin wrappers to adapt backbones with non-standard
+signatures (e.g., ``SongUNet``, ``DiT``) to the
```
Comment (megnvidia) [inline docs/api/diffusion/preconditioners.rst]:
```suggestion
signatures (for example, ``SongUNet``, ``DiT``) to the
```

File: docs/api/models/diffusion_unets.rst [hunk 10]
```diff
@@ -0,0 +1,325 @@
...
+described in :ref:`Model Backbones <diffusion_model_backbones>`.
+
+All models on this page are based on the
+:class:`~physicsnemo.core.module.Module` class.
+
+.. important::
+
+    These UNet backbones do **not** implement the
+    :class:`~physicsnemo.diffusion.DiffusionModel` protocol directly.  Their
+    forward signatures differ (e.g.,
```
Comment (megnvidia) [inline docs/api/models/diffusion_unets.rst:20]:
```suggestion
    forward signatures differ (for example,
```

File: docs/api/models/diffusion_unets.rst [hunk 11]
```diff
@@ -0,0 +1,325 @@
...
+:math:`\mathbf{x}(T + \Delta t)` (one time step forward), ..., up to :math:`\mathbf{x}(T + K \Delta t)`
+(K time steps forward). Such prediction horizons are called *lead-times* (a term
+adopted from the weather and climate forecasting community) and we want to apply
+diffusion to each of these latent states while accounting for their associated
+lead-time information.
+
+PhysicsNeMo provides a specialized architecture
+:class:`~physicsnemo.models.diffusion_unets.SongUNetPosLtEmbd` that implements
+lead-time aware models.  It extends ``SongUNet`` with learnable positional
+embeddings and lead-time embeddings: in its forward pass, the model uses the
```
Comment (megnvidia) [inline docs/api/models/diffusion_unets.rst]:
```suggestion
embeddings and lead-time embeddings. In its forward pass, the model uses the
```

File: docs/api/models/diffusion_unets.rst [hunk 12]
```diff
@@ -0,0 +1,325 @@
...
+    # lead-time labels 2, 0, 1 and concatenates them to the input x_cond before the first
+    # UNet block. In training mode, the model outputs logits for channels 1 and 3.
+    out = model(x_cond, noise_labels, class_labels, lead_time_label=lead_time_label)
+    print(out.shape)  # Shape: (B, C_x, res, res), same as the latent state
+
+    # If eval mode the model outputs probabilities for channels 1 and 3
+    model.eval()
+    out = model(x_cond, noise_labels, class_labels, lead_time_label=lead_time_label)
+
+.. note::
```
Comment (megnvidia) [inline docs/api/models/diffusion_unets.rst]: I have strong feelings that a note after a note should just be one note with two bullet items. I give you the option to choose the way you want to go.

File: docs/api/models/diffusion_unets.rst [hunk 13]
```diff
@@ -0,0 +1,325 @@
...
+    The ``SongUNetPosLtEmbd`` *is not* an autoregressive model that performs a rollout
+    to produce future predictions. From the point of view of the ``SongUNetPosLtEmbd``,
+    the lead-time information is *frozen*. The lead-time dependent latent state :math:`\mathbf{x}`
+    might however be produced by such an autoregressive/rollout model.
```
Comment (megnvidia) [inline docs/api/models/diffusion_unets.rst]:
```suggestion
    might however be produced by such an autoregressive or rollout model.
```

File: docs/api/diffusion/noise_schedulers.rst (hunk 3 above)
Comment (CharlelieLrt) [reply docs/api/diffusion/noise_schedulers.rst:31]: Yes, here the order matters

File: docs/api/diffusion/noise_schedulers.rst (hunk 4 above)
Comment (CharlelieLrt) [reply docs/api/diffusion/noise_schedulers.rst]: No, here the order does not matter. I'll change this to bullet-list

File: docs/api/diffusion/noise_schedulers.rst (hunk 2 above)
Comment (CharlelieLrt) [reply docs/api/diffusion/noise_schedulers.rst:17]: Yes, it does 😄

File: docs/api/diffusion/noise_schedulers.rst (hunk 5 above)
Comment (CharlelieLrt) [reply docs/api/diffusion/noise_schedulers.rst]: done

File: docs/api/diffusion/noise_schedulers.rst (hunk 6 above)
Comment (CharlelieLrt) [reply docs/api/diffusion/noise_schedulers.rst]: done

File: docs/api/diffusion/noise_schedulers.rst (hunk 7 above)
Comment (CharlelieLrt) [reply docs/api/diffusion/noise_schedulers.rst]: done

File: docs/api/diffusion/noise_schedulers.rst (hunk 8 above)
Comment (CharlelieLrt) [reply docs/api/diffusion/noise_schedulers.rst]: done

File: docs/api/diffusion/preconditioners.rst (hunk 9 above)
Comment (CharlelieLrt) [reply docs/api/diffusion/preconditioners.rst]: done

File: docs/api/models/diffusion_unets.rst (hunk 10 above)
Comment (CharlelieLrt) [reply docs/api/models/diffusion_unets.rst:20]: done

File: docs/api/models/diffusion_unets.rst (hunk 12 above)
Comment (CharlelieLrt) [reply docs/api/models/diffusion_unets.rst]: done
